include/taskolib/Sequence.h

```diff
@@ -706,6 +737,22 @@ class Sequence
     Iterator find_end_of_continuation(Iterator block_start);
     ConstIterator find_end_of_continuation(ConstIterator block_start) const;
 
+    /**
+     * Run a given execution function on the sequence, taking care of exception handling
+     * and messaging.
+     *
+     * \param context       Execution context
+     * \param comm_channel  Pointer to a communication channel; if null, messaging and
+     *                      cross-thread interaction are disabled.
+     * \param exec_block_name  Name of the execution block, preferably starting with a
+     *                      capital letter (e.g. "Sequence", "Step 10")
+     * \param fct           Function to be executed
+     */
+    void
+    handle_execution(Context& context, CommChannel* comm_channel,
+                     gul14::string_view exec_block_name,
+                     std::function<void(Context&, CommChannel*)> fct);
```

Comment:
No, this is internal. Unfortunately a function pointer does not suffice because we need captures. But we could probably turn this into a function template inside `Sequence.cc`. I am not sure if that is an overall gain. Performance-wise yes, but why would we care about a few cycles of overhead when launching a single step?

Reply:
But `handle_execution()` is always used, also for normal (Sequence-d) runs.

Ah I see it captures `this` (which we could probably get rid of) and `step_index` which is very hard to get rid of (i.e. impossiblish).

So .. ok.